File: Frontend/src/interfaces/userparams.ts
import { Member } from "./member";

export class userparams{
    gender:string ;
    minAge=18;
    maxage=100;
    pageNumber=1;
    pageSize=1;
    curentUsername:string;

    constructor(user:any){
     this.gender = user.gender=='male' ? 'female' : 'male';
     this.curentUsername= user.name;
    }
}

File: EmployeeBackend/Helpers/UserParams.cs
namespace EmployeeBackend.Helpers
{
    public class UserParams
    {
        public const int MaxPagedSize = 50;
        public int PageNumber { get; set; } = 1;

        private int _PageSize=10;

        public int PageSize
        {
            get => _PageSize;
            set=> _PageSize = (value>MaxPagedSize) ? MaxPagedSize : value;

        }

        public string? CurentUsername { get; set; }

        public string Gender { get; set; }

        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 100;

        public string? OrderBy { get; set; } = "LastActive";

    }
}
